src/record/record.js

```diff
@@ -431,7 +432,8 @@ Record.prototype._onRecordRecovered = function (remoteVersion, remoteData, messa
 
     const oldValue = this._$data
     const newValue = jsonPath.set(oldValue, undefined, data, false)
-    if (oldValue === newValue) {
+
+    if (utils.deepEquals(oldValue, remoteData)) {
       return
```

Comment:
This shouldn't be required... if `deepEquals` and `oldValue === newValue` differs then there is a bug in `jsonPath.set` and we should have a unit test for that added..